apps/reaper/lib/reaper/decoder/xml.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Reaper.Decoder.Xml do
+  @moduledoc """
+  Reaper.Decoder implementation to decode a csv file into a stream of records
+  """
+
+  alias Reaper.XmlSchemaMapper
+
+  defmodule XmlError do
+    defexception [:message]
+  end
+
+  @behaviour Reaper.Decoder
+
+  @impl Reaper.Decoder
+  def decode(
+        {:file, filename},
+        %SmartCity.Dataset{technical: %{schema: schema, topLevelSelector: top_level_selector}} = dataset
```

Comment:
Agreed, I just didn't come up with a better name at the time. I switched all uses of `top_level_selector` to `selector` since I think only the dataset really needs to care that it's the "topLevelSelector"